.ci/magician/exec/runner.go

```diff
@@ -0,0 +1,99 @@
+package exec
+
+import (
+	"container/list"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	cp "github.com/otiai10/copy"
+)
+
+type actualRunner struct {
+	cwd      string
+	dirStack *list.List
+}
+
+type Runner interface {
+	GetCWD() string
+	Copy(src, dest string) error
+	RemoveAll(path string) error
+	PushDir(path string) error
+	PopDir() error
+	WriteFile(name, data string) error
+	Run(name string, args, env []string) (string, error)
```

Comment:
RemoveAll
Copy
WriteFile still work from the default OS directory. We probably need to change to the CWD before executing them?

Reply:
I was intending for those to be used with complete paths. Should I make them return an error if a relative path is specified?

Reply:
I have made the runner use absolute paths for these three methods and also PushDir. Paths are assumed to be either absolute or relative to cwd.